### PR TITLE
Kristi

### PR DESCRIPTION
**Pull request by kristi-h**
Include `filteredData` in Context so the map can show updated filtered items, nest foodtype filter inside country filter and display final filtered results.

### CODE REVIEW

File: src/components/API/MenuResults.jsx [hunk 1]
```diff
@@ -27,59 +24,85 @@ export function MenuResults() {
       getOrder();
     }
 
+    const [data, setData] = useState([]);
```
Comment (yusuken1121) [inline src/components/API/MenuResults.jsx:27]: data and filteredData are declared twice, which will result in an error.

Comment (kristi-h) [reply src/components/API/MenuResults.jsx]: Thanks for pointing that out! Fixed and verified merged branch is running without errors!

File: src/components/molecules/filter/StateDropdown.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { useLocation } from "../../../contexts/FilterContext"
 
 export default function StateDropdown(){
 
-    const { selectState } = useLocation()
+    const { selectState, state } = useLocation()
 
     function selected(e){
         e.preventDefault()
```
Comment (yusuken1121) [inline src/components/molecules/filter/StateDropdown.jsx:9]: You might not need `preventDefault` for the `onChange` event.

**Review by yusuken1121 — APPROVED** [review]
I confirmed that the filtering by city is working and is reflected on the map!
LGTM!
My comments are just minor suggestions, so feel free to consider them for your next PR. I approved it :))

**Review by arositen — APPROVED** [review]
Looks good!

File: src/components/API/MenuResults.jsx [hunk 3]
```diff
@@ -7,16 +8,67 @@ import Rating from '@mui/material/Rating'
...
+            return ByFoodType(intermediateArray);
+        }
+
+        setFilteredData(FilterData())
+
+    }, [foodtype, state, city]);
+
+    function ByCountry() {
+
+        if (true) {
```
[the remark this answers is not in the document]
Comment (kristi-h) [reply src/components/API/MenuResults.jsx:41]: don't think so!

File: src/components/API/MenuResults.jsx [hunk 4]
```diff
@@ -7,16 +8,67 @@ import Rating from '@mui/material/Rating'
...
+        async function getResults() {
+            const menuData = await getMenuAPI();
+            setData(menuData);
+        }
+        getResults();
+    }, []);
 
-    const [order , setOrder] = useState([])
+    React.useEffect(() => {
+        // console.log("city", city);
```
[the remark this answers is not in the document]
Comment (kristi-h) [reply src/components/API/MenuResults.jsx:25]: ahh, this is not what I wrote so didn't want to just delete it.

File: src/components/molecules/filter/StateDropdown.jsx (hunk 2 above)
Comment (kristi-h) [reply src/components/molecules/filter/StateDropdown.jsx:9]: gotcha

**Pull request merged**